config/validate.go

```diff
@@ -3,24 +3,116 @@ package config
 import (
 	"fmt"
 	"github.com/BurntSushi/toml"
+	"github.com/agext/levenshtein"
+	"sort"
+	"strings"
 )
 
+// UnexpectedKey represents result of validation
+type UnexpectedKey struct {
+	Name        string
+	SuggestName string
+}
+
+func nameSuggestion(given string, suggestions []string) string {
+	for _, suggestion := range suggestions {
+		dist := levenshtein.Distance(given, suggestion, nil)
+		if dist < 3 {
+			return suggestion
+		}
+	}
+	return ""
+}
+
 // ValidateConfigFile detect unexpected key in configfile
-func ValidateConfigFile(file string) ([]string, error) {
+func ValidateConfigFile(file string) ([]UnexpectedKey, error) {
 	config := &Config{}
 	md, err := toml.DecodeFile(file, config)
 	if err != nil {
 		return nil, fmt.Errorf("failed to test config: %s", err)
 	}
 
-	var unexpectedKeys []string
+	var suggestions = []string{
```

Comment:
*reflect* package could get exported keys of *Config* type.
https://go.dev/play/p/BO67RHUMeXg